c/array/array_utils.c

```diff
@@ -71,36 +42,40 @@ void initialize_array_from_user(int *array, int size)
     }    
 }
 
-
 void initialize_array_with_random_elements(int *nums, int size)
 {
     for (int i = 0; i < size; i++)
-    { 
-        nums[i] = rand() ;
+    {
+        nums[i] = rand();
     }
 }
 
-int *clone_array(int *nums, int size)
+int* create_new_array_and_intialize_with_user_input(int size)
 {
-    return NULL;
+    int* nums = create_new_array(size);
+    initialize_array_from_user(nums, size);
+    return nums;
 }
 
-void rotate_array(int *nums, int size, int number_of_times_rotate)
-{
-    for (int count = 0; count < number_of_times_rotate; count++)
-    {
-        int last_element = nums[size - 1];
-        for (int i = size - 1; i > 0; i--)
-        {
-            nums[i] = nums[i - 1];
-        }
-        nums[0] = last_element;
-    }
+int* create_new_array_and_initalize_with_random_elements(int size)
+{    
+    int* array = create_new_array(size);
+    initialize_array_with_random_elements(array, size);
+    return array;
 }
 
-void print_array(int *nums, int size)
+int *clone_array(int *nums, int size)
+{   
+    int *clone_array = create_new_array(size);
+    memcpy(clone_array, nums, size * sizeof(int));
+    return clone_array;
+}
+
+
+
+void print_array(char name,int *nums, int size)
```

Comment:
void print_array(char *name,int *nums, int size)

for a string (charachter array should be "char*" )